From 68eaa07de69e873b89d4aba341c6ed1ca81f6819 Mon Sep 17 00:00:00 2001
From: Lubos Lunak <l.lunak@kde.org>
Date: Wed, 30 Jan 2008 14:24:01 +0100
Subject: [PATCH 03/16] This patch changes QObjectPrivateVersion, thus preventing mixing
 parts of upstream Qt and qt-copy. In general it is a bad idea to mix
 e.g. libQtCore from one build and libQtGui from another one, and other
 qt-copy patches could make changes in Qt internal structures that could
 cause problems when mixed with upstream Qt.

This patch does not make qt-copy binary incompatible with upstream Qt.
It only further enforces using the same sources for the whole Qt build.

qt-bugs@ issue : none
Trolltech task ID : none
bugs.kde.org number : none
---
 src/corelib/kernel/qobject_p.h |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

--- a/src/corelib/kernel/qobject_p.h
+++ b/src/corelib/kernel/qobject_p.h
@@ -82,7 +82,9 @@ void Q_CORE_EXPORT qt_register_signal_sp
 
 extern QSignalSpyCallbackSet Q_CORE_EXPORT qt_signal_spy_callback_set;
 
-enum { QObjectPrivateVersion = QT_VERSION };
+// add 0x1000000 to mark it as qt-copy version, with possible modifications
+// in some Q*Private class
+enum { QObjectPrivateVersion = QT_VERSION + 0x1000000 };
 
 class Q_CORE_EXPORT QAbstractDeclarativeData
 {
